### PR TITLE
Use Decompose for shared navigation

### PR DESCRIPTION
**Pull request by arkivanov**
As discussed on Twitter, here is a draft just to demonstrate how it would look like. Only the main Android app is working - it looks and feels basically the same, just added some fancy screen transitions. Wear, Auto (commented out), and any other platforms or tests are not touched yet. This should be also compatible with SwiftUI if needed. Unit and integration testing should be also pretty easy.

Tests and checks should be failing for now.

Please let me know your thoughts, feel free to discard if not interested, or I can finish the rest.

### CODE REVIEW

**Comment by arkivanov** [comment]
> is idea that Decompose would provide something similar instead?

Yes, there are two options. Either expose states via a `Flow` wrapper (like `CFlow`), or use `Value` provided by Decompose. IN the latter case, the SwiftUI side works as [explained in the docs](https://arkivanov.github.io/Decompose/component/overview/#swiftui-example).

**Comment by joreilly** [comment]
I guess more general question then is whether view models have a place when using Decompose?  Are there ways of using Decompose to specifically address shared navigation alone or is expectation that those broader changes would be needed as well to do that?

**Comment by arkivanov** [comment]
There is no such an entity as ViewModel in Decompose. In Jetpack Compose, the ViewModel plays two roles: 1. It retains in the back stack; 2. It retains over configuration changes.

With Decompose, the first case is handled automatically - components are retained in the back stack.

The second case can be [handled](https://arkivanov.github.io/Decompose/component/instance-retaining/) explicitly via `InstanceKeeper` if needed. However, in Confetti it is already handled by the app-scoped repository, so the app works just fine.

**Comment by arkivanov** [comment]
I have updated the iOS app.

**Comment by arkivanov** [comment]
Updated Android Auto app

**Comment by joreilly** [comment]
Not sure if it needs a rebase or something but getting following for some reason 

```
gh pr checkout 688
From https://github.com/joreilly/Confetti
 ! [rejected]          refs/pull/688/head -> decompose  (non-fast-forward)
failed to run git: exit status 1
```

**Comment by arkivanov** [comment]
Perhaps, that's because I force-pushed one of the commits

**Comment by joreilly** [comment]
ah, maybe I have branch here from before....let me delete it and try again.....yep, that was it!

**Comment by joreilly** [comment]
The failure before was related to jvm target and due to instance not running....should be addressed by https://github.com/joreilly/Confetti/pull/702

File: androidApp/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -73,8 +73,8 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
-<!--
+
```
Comment (yschimke) [inline androidApp/src/main/AndroidManifest.xml:76]: Should these be enabled in main?

Comment (arkivanov) [reply androidApp/src/main/AndroidManifest.xml:76]: Seems like I forgot to comment those back after testing Auto.

File: androidApp/build.gradle.kts [hunk 2]
```diff
@@ -155,9 +155,11 @@ dependencies {
     implementation(libs.compose.material)
     implementation(libs.compose.material.icons.core)
     implementation(libs.compose.material.icons.extended)
-    implementation(libs.compose.navigation)
     implementation(libs.coil.compose)
 
+    implementation(libs.decompose.decompose)
```
Comment (oianmol) [inline androidApp/build.gradle.kts:160]: do you need to `implement` this even if we have it specified it as `api` under shared's build.gradle.kts ?

`api(libs.decompose.decompose)` 🤔

Comment (arkivanov) [reply androidApp/build.gradle.kts:160]: Most likely we don't need! Thanks for spotting. Also we can look for unused dependencies in the shared module.

**Pull request merged**